rest/import_test.go

```diff
@@ -698,6 +698,236 @@ func TestMigrateWithExternalRevisions(t *testing.T) {
 	assert.Equals(t, len(doc.Meta.RevTree.BodyMap), 2)
 }
 
+// Write a doc via SDK with an expiry value.  Verify that expiry is preserved when doc is imported via DCP feed
+func TestXattrFeedBasedImportPreservesExpiry(t *testing.T) {
+
+	SkipImportTestsIfNotEnabled(t)
+
+	rt := RestTester{
+		SyncFn: `function(doc, oldDoc) { channel(doc.channels) }`,
+		DatabaseConfig: &DbConfig{
+			ImportDocs: "continuous",
+		},
+	}
+	defer rt.Close()
+	bucket := rt.Bucket()
+
+	rt.SendAdminRequest("PUT", "/_logging", `{"Import+":true, "CRUD+":true}`)
+
+	// 1. Create docs via the SDK with expiry set
+	mobileKey := "TestXattrImportPreservesExpiry"
+	mobileBody := make(map[string]interface{})
+	mobileBody["type"] = "mobile"
+	mobileBody["channels"] = "ABC"
+
+	// Write directly to bucket with an expiry
+	expiryUnixEpoch := time.Now().Add(time.Second * 30).Unix()
+	_, err := bucket.Add(mobileKey, uint32(expiryUnixEpoch), mobileBody)
+	assertNoError(t, err, "Error writing SDK doc")
+
+	// Wait until the change appears on the changes feed to ensure that it's been imported by this point
+	changes, err := rt.WaitForChanges(1, "/db/_changes", "", true)
+	assertNoError(t, err, "Error waiting for changes")
+
+	log.Printf("changes: %+v", changes)
+	changeEntry := changes.Results[0]
+	assert.Equals(t, changeEntry.ID, mobileKey)
+
+	// Double-check to make sure that it's been imported by checking the Sync Metadata in the xattr
+	assertXattrSyncMetaRevGeneration(t, bucket, mobileKey, 1)
+
+	// Verify the expiry has been preserved after the import
+	gocbBucket := bucket.(*base.CouchbaseBucketGoCB)
+	expiry, err := gocbBucket.GetExpiry(mobileKey)
+	assertNoError(t, err, "Error calling GetExpiry()")
+	assert.True(t, expiry == uint32(expiryUnixEpoch))
+
```

Comment:
Add a second case in here to validate that docs w/ no expiry are still working as expected (not setting expiry).

Reply:
Added